newIDE/app/src/UI/CompactSelectField/index.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+
+import * as React from 'react';
+import Tooltip from '@material-ui/core/Tooltip';
+import classNames from 'classnames';
+import classes from './CompactSelectField.module.css';
+import { tooltipEnterDelay } from '../Tooltip';
+type Props = {|
+  onChange: string => void,
+  value: string,
+  id?: string,
+  disabled?: boolean,
+  errored?: boolean,
+  placeholder?: string,
+  children: React.Node,
+  renderLeftIcon?: (className: string) => React.Node,
+  leftIconTooltip?: React.Node,
+|};
+
+const CompactSelectField = ({
+  onChange,
+  value,
+  id,
+  disabled,
+  errored,
+  placeholder,
+  children,
+  renderLeftIcon,
+  leftIconTooltip,
+}: Props) => {
+  return (
+    <div
+      className={classNames({
+        [classes.container]: true,
+        [classes.disabled]: disabled,
+        [classes.errored]: errored,
+      })}
+    >
+      {renderLeftIcon && (
+        <Tooltip
```

Comment:
are we always displaying a tooltip even if there is no tooltip?

Reply:
I added a check on the tooltip content. I think an icon should not be used without a tooltip